pytorch_lightning/accelerators/tpu.py

```diff
@@ -47,3 +47,7 @@ def get_device_stats(self, device: Union[str, torch.device]) -> Dict[str, Any]:
     def auto_device_count() -> int:
         """Get the devices when set to auto."""
         return 8
+
+    @staticmethod
+    def is_available() -> bool:
```

Comment:
for TPU I think it will fail early if no TPU will be available because we call `connect` before `setup_environment`.

Reply:
For TPU and IPU these checks might feel redundant because there's only a limited set of strategies the accelerator is  paired with. Someone may write a custom TPU training strategy, forget to add that check, but get covered by this one.